pkg/daemon/ceph/client/mgr.go

```diff
@@ -179,3 +189,17 @@ func ConfigureBalancerModule(context *clusterd.Context, clusterInfo *ClusterInfo
 
 	return nil
 }
+
+func getMinCompatClientVersion(clusterInfo *ClusterInfo, balancerModuleMode string) (string, error) {
+	// Set min compat client to luminous before enabling the balancer mode "upmap"
+	minCompatClientVersion := "luminous"
+	if balancerModuleMode == readBalancerMode || balancerModuleMode == upmapReadBalancerMode {
+		if !clusterInfo.CephVersion.IsAtLeast(cephver.CephVersion{Major: 18}) {
```

Comment:
Maybe I should just check if `squid` over here (ceph v19) if `upmap-read` and `read` will only be used in `squid`. I'll confirm that.